### PR TITLE
use UUID as the name of model directory

### PR DESCRIPTION
**Pull request by MengMeng96**
## Description

Name the model directory with UUID instead of the digest of the model file(model.mar). To distinguish between folders with different names but identical models. If name the directory with the digest of the model.mar, the function `TestUtils/unregisterModel()` will fail to delete the model directory(on windows 10) when there is another model use the same model.mar.

Fixes #(issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)

## Feature/Issue validation/testing

Please describe the tests [UT/IT] that you ran to verify your changes and relevent result summary. Provide instructions so it can be reproduced. 
Please also list any relevant details for your test configuration.

- [ ] frontend/server/src/test/java/org/pytorch/serve/ModelServerTest.java/testLoadModelWithInitialWorkersWithJSONReqBody()

- UT/IT execution results
PASSED

- Logs

## Checklist:

- [ ] Have you added tests that prove your fix is effective or that this feature works?
- [ ] New and existing unit tests pass locally with these changes?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

### CODE REVIEW

**Comment by MengMeng96** [comment]
> @MengMeng96 Can you please describe the real need for adding UUID support? TorchServe uses model names for the inference API, so the model names are expected to be unique across a deployment. If we replace the model names with UUID it will make it harder for people to use. Please explain the real purpose of this request. From the code changes it does not look like a blocking issue for adding Native Windows support for TorchServe.

In stead of `replace the model names with UUID`, I mean to replace the name of the directories which stores the models. At present, TorchServe name the directory with the digest of the model. The same models, e.g. noop, with different names, e.g. noop and noop_v1.0 are stored in the same directory. In windows, when I want to unregister noop, then the model files, e.g. model.pt and service.py, in the directory are deleted but the directory cannot be deleted which cause a failure in test case `testLoadModelWithInitialWorkersWithJSONReqBody`. If we name the directories whit UUID then each model has a unique directory and unregister a model will not affect other models. When I name the directories with UUID, the test case passed.

**Comment by dhaniram-kshirsagar** [comment]
@MengMeng96  I didn't get the meaning of `windows native`. Are you using Windows WSL to install this?

**Comment by MengMeng96** [comment]
> @MengMeng96 I didn't get the meaning of `windows native`. Are you using Windows WSL to install this?

No, not WSL, I'm trying to install TorchServe using Windows PowerShell.

**Comment by dhaniram-kshirsagar** [comment]
> > @MengMeng96 I didn't get the meaning of `windows native`. Are you using Windows WSL to install this?
> 
> No, not WSL, I'm trying to install TorchServe using Windows PowerShell.

NOTE - Officially, torchserve is certified using Win WSL and not Win PowerShell.

**Comment by MengMeng96** [comment]
> > > @MengMeng96 I didn't get the meaning of `windows native`. Are you using Windows WSL to install this?
> > 
> > 
> > No, not WSL, I'm trying to install TorchServe using Windows PowerShell.
> 
> NOTE - Officially, torchserve is certified using Win WSL and not Win PowerShell. https://github.com/pytorch/serve/blob/master/docs/torchserve_on_wsl.md

I know, and what I want to do is to use TorchServe on windows directly(not WSL).

**Comment by nbcsm** [comment]
@dhaniram-kshirsagar, we are trying to enable TorchServe on Windows.
We expected that all existing tests should pass.
But Test testLoadModelWithInitialWorkersWithJSONReqBody fails on windows, and this PR is to fix it.

**Comment by chauhang** [comment]
@dhaniram-kshirsagar Do you see any issues with making this change for how the .mar files are unzipped and saved in local for TorchServe? The fix and changes to the test cases are blocking Microsoft team from adding Native Windows support for TorchServe

**Comment by maaquib** [comment]
> @dhaniram-kshirsagar, we are trying to enable TorchServe on Windows.
> We expected that all existing tests should pass.
> But Test testLoadModelWithInitialWorkersWithJSONReqBody fails on windows, and this PR is to fix it.

@nbcsm Can you provide the stack trace? We can take a look

**Comment by MengMeng96** [comment]
@dhaniram-kshirsagar Hi, this is the stack trace of the test case testLoadModelWithInitialWorkersWithJSONReqBody.
[stacktrace.txt](https://github.com/pytorch/serve/files/5014315/stacktrace.txt)

**Comment by dhaniram-kshirsagar** [comment]
> @dhaniram-kshirsagar Hi, this is the stack trace of the test case testLoadModelWithInitialWorkersWithJSONReqBody.
> [stacktrace.txt](https://github.com/pytorch/serve/files/5014315/stacktrace.txt)

@MengMeng96 I have not tested this yet. I will look into it today.

File: frontend/modelarchive/src/main/java/org/pytorch/serve/archive/ModelArchive.java [hunk 1]
```diff
@@ -133,14 +133,10 @@ public static File unzip(InputStream is, String eTag) throws IOException {
         }
         ZipUtils.unzip(new DigestInputStream(is, md), tmp);
         if (eTag == null) {
-            eTag = HexUtils.toHexString(md.digest());
+            eTag = UUID.randomUUID().toString().replaceAll("-", "");
         }
+        logger.info("eTag {}", eTag);
```
Comment (maaquib) [inline frontend/modelarchive/src/main/java/org/pytorch/serve/archive/ModelArchive.java:138]: NIT: log message is not descriptive. We should probably add the model name to go along with the etag

Comment (MengMeng96) [reply frontend/modelarchive/src/main/java/org/pytorch/serve/archive/ModelArchive.java:138]: I printed the model name and this is the log.

TorchServeSuite > TorchServe > org.pytorch.serve.ModelServerTest > testLoadModelWithInitialWorkersWithJSONReqBody STANDARD_OUT
    2020-08-04 01:41:03,788 [DEBUG] nioEventLoopGroup-3-4 org.pytorch.serve.wlm.ModelVersionedRefs - Removed model: noop version: 1.11
    2020-08-04 01:41:03,794 [INFO ] nioEventLoopGroup-3-4 org.pytorch.serve.wlm.ModelManager - Model noop unregistered.
    2020-08-04 01:41:03,795 [INFO ] nioEventLoopGroup-3-4 ACCESS_LOG - /127.0.0.1:63658 "DELETE /models/noop HTTP/1.1" 200 7
    2020-08-04 01:41:03,795 [INFO ] nioEventLoopGroup-3-4 TS_METRICS - Requests2XX.Count:1|#Level:Host|#hostname:czhong03,timestamp:null
    2020-08-04 01:41:03,799 [INFO ] nioEventLoopGroup-3-4 org.pytorch.serve.wlm.ModelManager - modelName : noop
    2020-08-04 01:41:03,799 [INFO ] nioEventLoopGroup-3-4 org.pytorch.serve.archive.ModelArchive - marFileName : noop.mar
    2020-08-04 01:41:03,805 [INFO ] nioEventLoopGroup-3-4 org.pytorch.serve.archive.ModelArchive - model folder already exists: 29385dfc880480adb4ff8a26d9461d72539fb887
    2020-08-04 01:41:03,806 [INFO ] nioEventLoopGroup-3-4 ACCESS_LOG - /127.0.0.1:63658 "POST /models HTTP/1.1" 400 8
    2020-08-04 01:41:03,806 [INFO ] nioEventLoopGroup-3-4 TS_METRICS - Requests4XX.Count:1|#Level:Host|#hostname:czhong03,timestamp:null

TorchServeSuite > TorchServe > org.pytorch.serve.ModelServerTest > testLoadModelWithInitialWorkersWithJSONReqBody FAILED
    java.lang.AssertionError at ModelServerTest.java:317

Comment (maaquib) [reply frontend/modelarchive/src/main/java/org/pytorch/serve/archive/ModelArchive.java:138]:
```
./torchserve_sanity.sh | grep --line-buffered eTag
+ source scripts/install_utils
++ set -euo pipefail
+ set +u
+ install_torch_deps
+ '[' is_gpu_instance ']'
+ '[' '' == cuda101 ']'
+ pip install -U -r requirements.txt
+ set -u
+ install_pytest_suite_deps
+ pip install -U -r requirements/developer.txt
+ install_bert_dependencies
+ pip install transformers
+ run_backend_pytest
+ python -m pytest --cov-report html:htmlcov --cov=ts/ ts/tests/unit_tests/
+ echo 'Backend test suite execution successful'
+ build_frontend
+ frontend/gradlew -p frontend clean build
    2020-08-04 17:38:35,334 [INFO ] Test worker org.pytorch.serve.archive.ModelArchive - eTag 8652c3fcf1494865a7fb5126afc74e4a
    2020-08-04 17:38:36,527 [INFO ] epollEventLoopGroup-3-2 org.pytorch.serve.archive.ModelArchive - eTag cd453c8393b2476db54af1f3d544ec39
    2020-08-04 17:38:36,650 [INFO ] epollEventLoopGroup-3-2 org.pytorch.serve.archive.ModelArchive - eTag a9cf7eae14db4aa99da3e3238287adb6
    2020-08-04 17:38:36,748 [INFO ] epollEventLoopGroup-3-2 org.pytorch.serve.archive.ModelArchive - eTag 254257fef1704f879291a0383bf132e0
    2020-08-04 17:38:36,844 [INFO ] epollEventLoopGroup-3-2 org.pytorch.serve.archive.ModelArchive - eTag fbd18c2405d24ea18e188f4ffef89909
    2020-08-04 17:38:36,971 [INFO ] epollEventLoopGroup-3-2 org.pytorch.serve.archive.ModelArchive - eTag eb7bba80a9394835b36a31d0e70fe5bc
    2020-08-04 17:38:37,080 [INFO ] epollEventLoopGroup-3-2 org.pytorch.serve.archive.ModelArchive - eTag fd300918ceb14fddbc29358aad5db14d
    2020-08-04 17:38:37,180 [INFO ] epollEventLoopGroup-3-2 org.pytorch.serve.archive.ModelArchive - eTag 3d5f0fcf085e4934bed16e8bda5f6b7c
    2020-08-04 17:38:37,975 [INFO ] epollEventLoopGroup-3-2 org.pytorch.serve.archive.ModelArchive - eTag 23c3f2c81fd74e13b698e786be192b92
```

Comment (MengMeng96) [reply frontend/modelarchive/src/main/java/org/pytorch/serve/archive/ModelArchive.java:138]: > NIT: log message is not descriptive. We should probably add the model name to go along with the etag

`model name` doesn't passed to function ModelArchive.unzip() nor function ModelArchive.downloadModel(String modelStore, String url), so `add the model name to go along with the etag` need to change all the functions that us them. And use UUID is an easier and better way to distinguish these files. 

When I run `frontend/gradlew.bat -p frontend clean build` on Windows 10, there are 3 test cases using the same eTag 29385dfc880480adb4ff8a26d9461d72539fb887, here is the log https://paste.ubuntu.com/p/sBjJmnWDSv/

**Comment by dhaniram-kshirsagar** [comment]
> @dhaniram-kshirsagar Do you see any issues with making this change for how the .mar files are unzipped and saved in local for TorchServe? The fix and changes to the test cases are blocking Microsoft team from adding Native Windows support for TorchServe

@chauhang We don't see any issues with this change.

**Comment by dhaniram-kshirsagar** [comment]
> > @dhaniram-kshirsagar Hi, this is the stack trace of the test case testLoadModelWithInitialWorkersWithJSONReqBody.
> > [stacktrace.txt](https://github.com/pytorch/serve/files/5014315/stacktrace.txt)
> 
> @MengMeng96 I have not tested this yet. I will look into it today.

@

> @dhaniram-kshirsagar, we are trying to enable TorchServe on Windows.
> We expected that all existing tests should pass.
> But Test testLoadModelWithInitialWorkersWithJSONReqBody fails on windows, and this PR is to fix it.

@MengMeng96 @nbcsm Tried to build with win PowerShell however for me **around 57 test cases are failing**.
Looks like few are related to 32bit vs 64bit numpy issues. Will correct those issues and update on this.

**Pull request merged**